account_invoice_temporal/models/account_invoice.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# © 2015 Esther Martín - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import _, api, exceptions, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def check_temporal(self):
+        if self.invoice_line.filtered('temporal'):
+            raise exceptions.Warning(_('The account is not assigned'))
+        else:
+            self.invoice_validate()
```

Comment:
En principio, también existe un asistente que te permite pasar de borrador a validado varias facturas a la vez. Pero por otro lado, no estoy segura de que se pueda lanzar una señal de workflow para varios registros a la vez. Por lo que seguramente habría que hacer un for ahi, o al menos mirar como ejecuta eso el asistente, igual con poner self.ensure_one aqui sería suficiente, porque el for se hace en el asistente. De todas formas, no estoy 100% segura.